cli/registry/client/endpoint.go

```diff
@@ -63,7 +69,7 @@ func getDefaultEndpointFromRepoInfo(repoInfo *registry.RepositoryInfo) (registry
 	}
 	// Default to the highest priority endpoint to return
 	endpoint := endpoints[0]
-	if !repoInfo.Index.Secure {
+	if insecure {
 		for _, ep := range endpoints {
 			if ep.URL.Scheme == "http" {
```

Comment:
I'm actually wondering if this original implementation is correct; looking at this code, it seems to _always_ use `http`, instead of _trying_ https and _falling back_ to `http`.

`--insecure` / `--insecure-registry` _should_ mean;

- try https
- if the certificate is self-signed, allow
- if https failed, fall back to http

Reply:
Oh, maybe it does, because `registryService.LookupPushEndpoints` seems to check if it can find certificates for the host, but please double check if it's doing the right thing

Reply:
1. when `if !repoInfo.Index.Secure {`, it always use `https`, because `!repoInfo.Index.Secure` is always `false`;
2. when `if insecure {`, it always use `http`, 
3. the result of `registryService.LookupPushEndpoints` is `endpoints: [{false https://myregistry-other.domain.com v2 false false true 0xc0004c5800} {false http://myregistry-other.domain.com v2 false false true 0xc0004c5800}]`